docs/releases.md

```diff
@@ -0,0 +1,80 @@
+
+Depending on the release, releasing and testing an RC may not be necessary (i.e. skip step 2); for larger releases, multiple RCs may be worthwhile (i.e. repeat steps 1-3).
+
+### 1 – Prepare code & build zip file
+
+- Confirm all work intended for release (fixes, features) is completed and merged to release branch.
+- Ensure your local checkout is in release branch and up to date!
+- Update version numbers and release date:
+  - `readme.txt`
+  - `style.scss`
+  - `package.json` and `package-lock.json`
+- Confirm/update metadata in `readme.txt`, e.g. “tested up to” version.
+- Draft changelog and add to `readme.txt`.
+- Clean install of dependencies: `npm ci`
+- Run a production build: `npm run deploy`. Note this does not deploy anywhere, you can run multiple times safely 🙂
+- Ensure all changes above are committed and pushed (you should not have uncommitted changes):
+  - Version numbers, dates, metadata
+  - Changelog
+  - Translation (`.pot`) file. Note this file is generated by production build.
+
+__*Outcome*: A `storefront.zip` file that you can test with.__ 
+
+### 2 – Publish RC build (optional)
+
+- Publish release on [GitHub](https://github.com/woocommerce/storefront/releases).
+  - Upload release zip.
+  - Use this format for tag: `version/1.2.3-rc.1`.
+  - Paste changelog into “release details” field.
+
+__*Outcome*: A `storefront.zip` file is available to the community and other stakeholders for testing and feedback.__
+
+### 3 – Test & QA
+
+Note: all new code should have been fully tested during development. This test pass is to confirm that there are no major regressions or bugs, aka [“happy path”](https://en.wikipedia.org/wiki/Happy_path) or [smoke testing](http://softwaretestingfundamentals.com/smoke-testing/).
+
+- __Recommended__: Test in a clean environment, similar to a typical hosting environment. Avoid testing in your development environment.
+- __Recommended__: Test with a snapshot of data from a real store, with customers, products and reviews.
+- Test core WordPress and Woo flows.
+- Test core Storefront features.
+- Test compatibility with Storefront child themes and plugins.
+- Spot-check to confirm recent fixes (this release, previous release) are working correctly and haven’t regressed.
+- If there are blocking issues or major regressions, the process stops here! (So they can be fixed.)
+
+__*Outcome*: You (release lead) are confident that the product is ready for release, and a safe & worthwhile upgrade for merchants.__
+
+### 4 – Launch release
+
+Now we get to release!
+
+- If necessary, build final `storefront.zip` with correct (non-RC) version number (repeat step 1).
+- Publish release on [GitHub](https://github.com/woocommerce/storefront/releases).
+  - Upload release zip.
+  - Use this format for tag: `version/1.2.3`.
+  - Paste changelog into “release details” field.
+- Upload to [WordPress.org](https://wordpress.org/themes/upload/). (You'll need access to the `Automattic` user account.)
```

Comment:
Can't we use [WordPress.org Plugin Deploy workflow](https://github.com/10up/action-wordpress-plugin-deploy) from 10up to deploy new releases from GitHub to WordPress.org without manually moving files?

Reply:
That's a possibility, though it's not what we currently do. This PR is focused on documenting the existing process - we can look at improvements or changes as follow up.

Note that Storefront has a build process to generates `.css` files from `.scss` (etc), so the tagged version of the source code doesn't actually match the built theme. So it's important that we upload the built zip to WordPress.org. I'm not familiar with the 10up workflow but it may not apply directly to how this repo works.

Reply:
Sounds good. I'll add the plugin deployment workflow as a separate ticket that could be played in the future.